src/sema/ast.rs

```diff
@@ -838,8 +838,9 @@ pub enum Expression {
     },
     StructLiteral {
         loc: pt::Loc,
+        id: pt::IdentifierPath,
         ty: Type,
-        values: Vec<Expression>,
+        values: Vec<(Option<pt::Loc>, Expression)>,
```

Comment:
Needs a comment explaining that the `pt::Loc` is the location of the field name, if the literal was specified with field names.

Reply:
It might make more sense to have `Vec<(Option<pt::Identifier>, Expression)>`, that is much clearer